### ISSUE TITLE
Sparse Categorical CrossEntropy in Google Collab

### ISSUE THREAD

**Issue by zerowu49** [tensorflow/tensorflow#39801]
https://pastebin.com/efb18WPB
I've been in trouble when configure the model.compile and model.fit in Google Collaboratory.
I want my model in
- Divided into train set and validation set
- Implement image augmentation
- Use Image Data Generator
- Sequential Model and using more than 1 hidden layer
- Training the model in max 30 minutes
- Accuracy min 85%
- Could predict picture if we upload it.

I've been new in this ML. Please help me ASAP. Thankyou for your participation.

**Comment by abhipn** [comment]
@zerowu49 As far your code is concerned, there are two changes you need to make here, one is change the loss function,

`loss_fn = tf.keras.losses.SparseCategoricalCrossentropy()`

And another change would be to change the input_shape from `(150, 150, 3)` to `(200, 250, 3)` because in your data generator you have specified your target shape as `(200,, 250)`

`tf.keras.layers.Conv2D(32, (3,3), activation='relu', input_shape=(200, 250, 3))`

**Comment by amahendrakar** [comment]
@zerowu49,
Please check @abhipn's comment and let us know if it works.

I was able to run the code without any issues after making the suggested changes, please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/de056e4a4acc9dae035cb8aa635cf915/39801.ipynb). Thanks!

**Comment by zerowu49** [comment]
> 
> 
> @zerowu49 As far your code is concerned, there are two changes you need to make here, one is change the loss function,
> 
> `loss_fn = tf.keras.losses.SparseCategoricalCrossentropy()`
> 
> And another change would be to change the input_shape from `(150, 150, 3)` to `(200, 250, 3)` because in your data generator you have specified your target shape as `(200,, 250)`
> 
> `tf.keras.layers.Conv2D(32, (3,3), activation='relu', input_shape=(200, 250, 3))`

Thankyou, the model works. But how to optimize the model here?

**Comment by abhipn** [comment]
@zerowu49 One major change you need to make would be not use data augmentation techniques on validation dataset, you should only do that for train data. You can use rescale `normalization` for validation data like below,

`
test_datagen = ImageDataGenerator(rescale=1./255,)
`

Apart from that, try out different filter sizes, layers, activations, learning rate, epochs, batch size etc., to optimize your model. You have to try and find best hyperparameters that results in best accuracy. There's no fixed correct answer here. Good luck.

**Comment by amahendrakar** [comment]
@zerowu49,
Is this still an issue? Please feel free to close the issue if resolved. Thanks!
